### PR TITLE
Adds printable medipens

### PR DESCRIPTION
**Pull request by St0rmC4st3r**
## About The Pull Request

image

Adds printable epinephrine, atropine and dexaline medipens to the medical techfab after biotech research.

On a side note, reagent requirements for techfab designs no longer scale with efficiency. This should not be really an issue since only two other techfab designs use reagents.
This is done because otherwise this would allow reagent duping, when printing medipens even in basic configuration, but becoming more severe with upgrades. (On maximum efficiency atropine medipens would cost 4u of atropine to make despite containing 10u)

## Why It's Good For The Game

It seems to be a good compromise between adding medipen creators for every chemical and not being able to make them at all.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Trust me, it works. I have tested it.

</details>

## Changelog
:cl:
add: Epinephrine, Dexalin and Atropine medipens are now printable in the medical lathe.
tweak: Reagents in the protolathe recipes no longer scale with part efficiency.
/:cl:

### CODE REVIEW

**Comment by EvilDragonfiend** [comment]
I thought it meant a custom medipen, but I noticed how it works. very good.

**Comment by Rukofamicom** [comment]
I'm not 100% sure about printable atropine pens. Atropine is already absurdly strong as both a curative and for offensive use without instant injectors that make it easy to overdose someone. I'm aware the injectors already exist, but I removed them from medbay vendors a while back for this combination of reasons.

That said, I do think this would be a great way to add a lot more types of medipens that don't currently exist because they definitely deserve to have a place with more reagents.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Update the changes, explain your changes to removing the lathe efficiency from reagent cost.

**Comment by PepegaSandwich** [comment]
Why not simply make chemmaster subcategory of Injector?

**Comment by williamcll** [comment]
how about instead print empty injectors but you can put them into a chemical dispenser/mixer?

**Review by PowerfulBacon — APPROVED** [review]

File: code/modules/research/designs/medical_designs.dm [hunk 1]
```diff
@@ -192,6 +192,39 @@
...
 	departmental_flags = DEPARTMENTAL_FLAG_MEDICAL
 
+/datum/design/medipen_epi
+	name = "Epinephrine Medipen"
+	desc = "A rapid and safe way to stabilize patients in critical condition for personnel without advanced medical knowledge. Contains a powerful preservative that can delay decomposition when applied to a dead body."
+	id = "medipen_epi"
+	build_path = /obj/item/reagent_containers/hypospray/medipen
+	build_type = PROTOLATHE
+	materials = list(/datum/material/plastic = 200)
+	reagents_list = list(/datum/reagent/medicine/epinephrine = 10, /datum/reagent/toxin/formaldehyde = 3)
```
Comment (Archanial) [inline code/modules/research/designs/medical_designs.dm:202]: Making techfabs produce reagents is a bad idea.

**Comment by ToasterBan** [comment]
>how about instead print empty injectors but you can put them into a chemical dispenser/mixer?

>Why not simply make chemmaster subcategory of Injector?

because that would let you have access to effectively infinite ghetto sleepy pens/hyposprays

instant injection with only a small message in chat and without showing you who did it is very powerful

File: code/modules/research/designs/medical_designs.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/modules/research/designs/medical_designs.dm:202]: This means it requires the reagent, not that it produces it.

Comment (St0rmC4st3r) [reply code/modules/research/designs/medical_designs.dm:202]: > Making techfabs produce reagents is a bad idea.

It doesn't produce the reagents. It technically could before I made the reagent cost not scale with efficiency, but not anymore.

**Comment by St0rmC4st3r** [comment]
> how about instead print empty injectors but you can put them into a chemical dispenser/mixer?

This idea was raised many times. Too powerful for a regular crewmember. Shotgun darts are fired from a shotgun, syringes are fired from a syringe gun and can be removed, both can be countered by thick clothing, and medipens work through even hardsuits.

**Comment by PepegaSandwich** [comment]
> > how about instead print empty injectors but you can put them into a chemical dispenser/mixer?
> 
> This idea was raised many times. Too powerful for a regular crewmember. Shotgun darts are fired from a shotgun, syringes are fired from a syringe gun and can be removed, both can be countered by thick clothing, and medipens work through even hardsuits.

Anything can be op. By applying this logic we should nerf everything because mah I ded PRs.
Just because there is possibility of people abusing custom pens , it doesnt mean this feature idea is bad. You can OD people with regular pens ez way. Just make EMAG locks on pens so they cant be filled with harmful things. Add special sprite overlay for them to be spottable, or better, 0.5 seconds delay for injection, to prevent instant chain injecting. Also add delay for hardsuit injecting because *you are searching for injection port on John'Sex's Hardsuit...*

**Pull request merged**